nuts/client/authorizer.go

```diff
@@ -0,0 +1,25 @@
+package client
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/nuts-foundation/nuts-demo-ehr/keyring"
+)
+
+// RequestEditorFn defines the type of function used to modify outgoing HTTP requests
+type RequestEditorFn func(ctx context.Context, req *http.Request) error
+
+// Authorizer authorizes requests to nuts-node endpoints using the given private key
+type Authorizer struct {
+	Key keyring.Key
+}
+
+// RequestEditorFn returns a RequestEditorFn suitable for use in WithRequestEditorFn
+// of HTTP client constructors.
+func (a *Authorizer) RequestEditorFn() RequestEditorFn {
+	return func(ctx context.Context, req *http.Request) error {
+		req.Header.Set("Authorization", "foo")
```

Comment:
this doesn't seem right, have you tested it?

Reply:
This was a placeholder while I figured out the structure of things for the lib functions. It's been updated

Reply:
Then the jwt generator is going to be using the new lib, or is it going to be removed altogether?